### PR TITLE
chore(testing/sampletests): make its own module

### PR DESCRIPTION
**Pull request by codyoss**
With the most recent commit, x/tools/packages will require 1.12.
Carve out a new module with proper go version.

### CODE REVIEW

**Comment by codyoss** [comment]
Related to https://github.com/googleapis/google-cloud-go/pull/2849

File: testing/sampletests/go.mod [hunk 1]
```diff
@@ -0,0 +1,9 @@
+module github.com/GoogleCloudPlatform/golang-samples/testing/sampletests
+
+go 1.12
+
+require (
+	github.com/google/go-cmp v0.5.2
+	github.com/jstemmer/go-junit-report v0.9.1
```
Comment (tbpg) [inline testing/sampletests/go.mod:7]: I think there is a replace in the root go.mod we should copy over here.

**Review by tbpg — CHANGES_REQUESTED** [review]
Please double-check the installation of sampletests just in case we need to `cd` into that directory first (so we are inside the right module).

File: testing/sampletests/go.mod (hunk 1 above)
Comment (codyoss) [reply testing/sampletests/go.mod:7]: Whoops, good catch!

**Comment by codyoss** [comment]
> Please double-check the installation of sampletests just in case we need to cd into that directory first (so we are inside the right module).

You're right, fixed.

**Review by tbpg — APPROVED** [review]

**Pull request merged**